packages/slate/src/changes/at-range.js

```diff
@@ -863,7 +863,7 @@ Changes.insertTextAtRange = (change, range, text, marks, options = {}) => {
   }
 
   // PERF: Unless specified, don't normalize if only inserting text.
-  if (normalize !== undefined) {
+  if (normalize === undefined) {
```

Comment:
Now that normalize uses `getFlag` and can't be undefined anymore, what should this condition be?

1. `if (normalize)` would ignore `{normalize: true}` for one-line changes, which sounds counter-intuitive to me (the user expects its options to be taken into account)
2. `if (!normalize)` would not trigger the optimization unless `{normalize: false}` is specified
3. `if (options.normalize === undefined)` sounds like the correct behaviour to me (we optimize the normalization unless the user wants a specific behaviour), but I'm not sure it respects the code-base guidelines?

Hopefully we could merge this PR once this is settled?

Reply:
(Hi, DamareYoh is my non-work alt so I made the original suggestion).

What if we reverted the `normalize = change.getFlag('normalize', options)` to what it was before, 

and then change the `range.isExpanded` usage to:

```
if (normalize === undefined) {
   normalize = change.getFlag('normalize', options) && range.isExpanded;
} 
```

Or something like that. This approach would 

1. Maintain the original behavior where the default is to not normalize the insertion of a text character when the range is collapsed
2. Maintain the original behavior where the user could override the default behavior specifically by passing in an explicit `{ normalize: true }` option
3. Utilize the `getFlag('normalize', options)` behavior so that any logic that turns off normalization external to the function is appropriately utilized

Reply:
Sounds like a very good alternative solution! 
I'll fix the PR with your suggestion asap.

Reply:
Cool! One thing id suggest though is making some tests to verify. The logic here is complicated enough to be a source of mistakes and has been in the past, as we can tell :)

Reply:
I strongly agree we shall have some tests about that.  Even we fix this this time, we may carelessly disturb that in future PRs.  We shall have tests to ensure `normalize` of insertTextAtRange is and will be always working correctly.